backend/app/routers/visualization.py

```diff
@@ -0,0 +1,129 @@
+        file = await FileDB.get(PydanticObjectId(resource_id))
+        if file is not None:
+            resource_ref = MongoDBRef(
+                collection="files", resource_id=vizconfig_in["resource_id"]
+            )
+            viz_config = VizConfigDB(**vizconfig_in, resource=resource_ref)
+            await viz_config.insert()
+            return viz_config.dict()
+        else:
+            raise HTTPException(status_code=404, detail=f"File {resource_id} not found")
+    elif vizconfig_in["resource_type"] == "dataset":
+        dataset = await DatasetDB.get(PydanticObjectId(resource_id))
+        if dataset is not None:
+            resource_ref = MongoDBRef(
+                collection="datasets", resource_id=vizconfig_in["resource_id"]
+            )
+            viz_config = VizConfigDB(**vizconfig_in, resource=resource_ref)
+            await viz_config.insert()
+            return viz_config.dict()
+        else:
+            raise HTTPException(
+                status_code=404, detail=f"Dataset {resource_id} not found"
+            )
+
+
+@router.get("/{file_id}/config_file", response_model=List[VizConfigOut])
+async def get_resource_vizconfig(
+    file_id: PydanticObjectId,
+    user=Depends(get_current_user),
+    credentials: HTTPAuthorizationCredentials = Security(security),
+):
+    if (file := await FileDB.get(PydanticObjectId(file_id))) is not None:
+        query = [VizConfigDB.resource.resource_id == ObjectId(file_id)]
+        vizconfigs = []
+        async for vzconfig in VizConfigDB.find(*query):
+            vizconfigs.append(vzconfig)
+        return [vz.dict() for vz in vizconfigs]
+    else:
+        raise HTTPException(status_code=404, detail=f"File {file_id} not found")
+
+
+@router.get("/{dataset_id}/config_dataset", response_model=List[VizConfigOut])
+async def get_resource_vizconfig(
+    dataset_id: PydanticObjectId,
+    user=Depends(get_current_user),
+    credentials: HTTPAuthorizationCredentials = Security(security),
+):
+    if (dataset := await DatasetDB.get(PydanticObjectId(dataset_id))) is not None:
+        query = [VizConfigDB.resource.resource_id == ObjectId(dataset_id)]
+        vizconfigs = []
+        async for vzconfig in VizConfigDB.find(*query):
+            vizconfigs.append(vzconfig)
+        return [vz.dict() for vz in vizconfigs]
+    else:
+        raise HTTPException(status_code=404, detail=f"Dataset {dataset_id} not found")
+
+
+@router.get("/config/{config_id}", response_model=VizConfigOut)
+async def get_vizconfig(
+    config_id: PydanticObjectId,
+    user=Depends(get_current_user),
+    credentials: HTTPAuthorizationCredentials = Security(security),
+):
+    if (viz_config := await VizConfigDB.get(PydanticObjectId(config_id))) is not None:
+        return viz_config.dict()
+    else:
+        raise HTTPException(status_code=404, detail=f"VizConfig {config_id} not found")
+
+
+@router.patch("/config/{config_id}/map", response_model=VizConfigOut)
```

Comment:
just curious what's this endpoint for?

Reply:
this is for when the vizualization is something like a URL - the way that an openlayers GIS preview works. It probably needs a better name or maybe I can just call it vizualization_data?

The map was because it should be key value pairs.